qemu/tests/multi_disk.py

```diff
@@ -106,28 +108,52 @@ def _do_post_cmd(session):
             session.cmd_status_output(cmd)
         session.close()
 
-    def _get_disk_index(session, image_size, disk_indexs):
-        list_disk_cmd = "echo list disk > disk && "
-        list_disk_cmd += "echo exit >> disk && diskpart /s disk"
-        disks = session.cmd_output(list_disk_cmd)
-        size_type = image_size[-1] + "B"
-        disk_size = ""
-
-        if size_type == "MB":
-            disk_size = image_size[:-1] + " MB"
-        elif size_type == "GB" and int(image_size[:-1]) < 8:
-            disk_size = str(int(image_size[:-1])*1024) + " MB"
-        else:
-            disk_size = image_size[:-1] + " GB"
+    def _get_disk_index(session, image_size, black_list, re_str):
+        """
+        Get disks ID list in guest. For linux guest, it's disk
+        kname; for windows guest, it's disk index which shows in
+        'diskpart list disk'
+        """
+        list_disk_cmd = params.get("list_disk_cmd")
+        disks = session.cmd_output(list_disk_cmd, timeout=120)
 
-        regex_str = 'Disk (\d+).*?%s.*?%s' % (disk_size, disk_size)
-        for disk in disks.splitlines():
-            if disk.startswith("  Disk"):
+        # Preprocessing
+        if params.get("os_type") == "windows":
+            size_type = image_size[-1] + "B"
+            disk_size = ""
+            if size_type == "MB":
+                disk_size = image_size[:-1] + " MB"
+            elif size_type == "GB" and int(image_size[:-1]) < 8:
+                disk_size = str(int(image_size[:-1])*1024) + " MB"
+            else:
+                disk_size = image_size[:-1] + " GB"
+        else:
+            output = session.cmd("mount")
+            re_str = params["re_str"]
+            exist_mountpoints = re.findall(r"^/dev/(%s)\d*" % re_str, output,
+                                           re.M)
+            reg_str = "^%s\d+|^dm-\d+" % re_str
+            exist_partitions = re.findall(reg_str, disks, re.M)
+            black_list.extend(exist_mountpoints)
+            black_list.extend(exist_partitions)
+
+        disks_splitlines = list(disks.splitlines())
+        disks_splitlines.pop(0)
+        disk_indexs = []
+        for disk in disks_splitlines:
+            if params.get("os_type") == "windows":
```

Comment:
Instead of reading `os_type` from params each and every time, read once and keep it in a variable and use it.